File: java/program/Clothing.java
// deklarasi class
class Clothing extends Product {

    // attibut private bertipe data string
    private String size;
    private String material;
    private String gender;

    // constructor tanpa parameter
    public Clothing() {
        this.size = "";
        this.material = "";
        this.gender = "";

    }

    // constructor dengan parameter size, nama, gender
    public Clothing(String size, String material, String gender) {
        this.size = size;
        this.material = material;
        this.gender = gender;
    }

    // setter dan getter untuk setiap atribut size, nama, gender
    public String get_size() {
        return this.size;
    }

    public void set_size(String size) {
        this.size = size;
    }

    public String get_material() {
        return this.material;
    }

    public void set_material(String material) {
        this.material = material;
    }

    public String get_gender() {
        return this.gender;
    }

    public void set_gender(String gender) {
        this.gender = gender;
    }

};

File: c++/program/Clothing.cpp
// import library
#include <iostream>
#include <string>
#include "Product.cpp"

using namespace std;

// deklarasi kelas
class Clothing : public Product
{

    // attibut private bertipe data string
private:
    string size;
    string material;
    string gender;

    // public Methods
public:
    // constructor tanpa parameter
    Clothing()
    {
        this->size = "";
        this->material = "";
        this->gender = "";
    }

    // constructor
    Clothing(string size, string material, string gender)
    {
        this->size = size;
        this->material = material;
        this->gender = gender;
    }

    // setter dan getter
    string get_size()
    {
        return this->size;
    }

    void set_size(string size)
    {
        this->size = size;
    }

    string get_material()
    {
        return this->material;
    }

    void set_material(string material)
    {
        this->material = material;
    }

    string get_gender()
    {
        return this->gender;
    }

    void set_gender(string gender)
    {
        this->gender = gender;
    }

    // deconstructor
    ~Clothing()
    {
    }
};
